internal/index/autoindex.go

```diff
@@ -24,26 +24,49 @@ import (
 )
 
 type autoIndexer struct {
-	watcher *watcher
+	done      chan struct{}
+	perFileFn func(string)
```

Comment:
Maybe we should just discard ``perFileFn``  and ``perDirFn`` as a variables and replace them with normal methods or inline the behavior where these functions are used since they are never changed and only assigned inline in the ``NewAutoIndexer`` function?

Reply:
Never mind, see they changed based on Watch

Reply:
I think we should name the functions ``fileFilterFn`` or ``filePredicateFn`` and ``dirFilterFn`` to reflect their purpose more

Reply:
Don't agree. `perFilterFn` filters AND "schedules" a file. `perDirFn` filters AND (if watch is enabled) adds a dir to be watched.